### PR TITLE
Customer swagger implemented #40

### PR DESCRIPTION
**Pull request by afrakucukaydin**

### CODE REVIEW

File: src/routes/customer.js [hunk 1]
```diff
@@ -1,15 +1,124 @@
...
+ * /customer:
+ *   post:
+ *     tags: [Customer]
+ *     summary: Add a new customer
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/Customer'
```
Comment (radmanlo) [inline src/routes/customer.js:39]: You should create a schema for the customer in the configuration/swagger.js. look at the swagger.js you will find out

File: src/routes/customer.js [hunk 2]
```diff
@@ -1,15 +1,124 @@
...
+ *         name: id
+ *         required: true
+ *         schema:
+ *           type: string
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/schemas/Customer'
```
Comment (radmanlo) [inline src/routes/customer.js:89]: Also for updates you need to create a Schema in swagger.js

**Review by radmanlo — CHANGES_REQUESTED** [review]

File: src/configuration/swagger.js [hunk 3]
```diff
@@ -2,48 +2,80 @@ const swaggerJsdoc = require('swagger-jsdoc');
...
+          properties: {
+            dish_name: {
+              type: 'string',
+            },
+            dish_ingredient: {
+              type: 'string',
+            },
           },
+          required: ['dish_name', 'dish_ingredient'],
+        },
```
Comment (Eng-NUREDDIN) [inline src/configuration/swagger.js:40]: please pull the updates from the main branch first there are many schemas missing

File: src/configuration/swagger.js [hunk 4]
```diff
@@ -2,48 +2,140 @@ const swaggerJsdoc = require('swagger-jsdoc');
...
+        Customer: {
+          type: 'object',
+          properties: {
+            customer_email: {
+              type: 'string',
+            },
+            customer_name: {
+              type: 'string',
+            },
+            created_at: {
```
Comment (radmanlo) [inline src/configuration/swagger.js:110]: Created_at has a default value which is now so you do not need to ask for that from the user.

File: src/configuration/swagger.js [hunk 5]
```diff
@@ -2,48 +2,140 @@ const swaggerJsdoc = require('swagger-jsdoc');
...
+              format: 'date-time',
+            },
+            customer_address: {
+              type: 'string',
+            },
+            birthdate: {
+              type: 'string',
+              format: 'date',
+            },
+            provider: {
```
Comment (radmanlo) [inline src/configuration/swagger.js:121]: for now, we do not need provider and provider_id please remove these 2

File: src/configuration/swagger.js [hunk 6]
```diff
@@ -2,48 +2,140 @@ const swaggerJsdoc = require('swagger-jsdoc');
...
+              type: 'string',
+            },
+            provider_id: {
```
Comment (radmanlo) [inline src/configuration/swagger.js:124]: for now, we do not need provider and provider_id please remove these 2

**Review by radmanlo — CHANGES_REQUESTED** [review]

**Review by radmanlo — APPROVED** [review]

**Review by Eng-NUREDDIN — APPROVED** [review]

**Review by talal-bakkour — APPROVED** [review]

**Pull request merged**